### ISSUE TITLE
Active Directory issues

### ISSUE THREAD

**Issue by thepaleone** [dubeaud/bugnet#132]
We've been using AD to authenticate with BugNET for a long time with no problems. Recently we had some new folks that needed to access our BN database. When attempting to log on, it simply fails with an unauthorized and does not automatically create the BN user like it used to do.

I can't find anything in any error logs or figure out why it's doing this. I tried to debug it on my local machine but it seems to be failing prior to getting to any BN related code.

My web.config is configured thusly:
```xml
<authentication mode="Windows">    
</authentication>
<authorization>
    <deny users="?" />
</authorization>
```
In BugNET I have the following settings
User Account Source: Active Directory
User Registration: None (I also tried it with Verified)
Domain / path: My domain name (should this be an AD path to the users?)
Username: <valid user>
Password: <verified password>
Anonymous Access: Disabled

We haven't had a new user for a long time, probably at least 3 versions of BugNET. Any ideas what I need to do to make this work again?

Thanks in advance

**Comment by dubeaud** [comment]
I would double check that the authentication module is uncommented in the web.config first thing.

**Comment by thepaleone** [comment]
It was commented. I removed the comments and tried again but the result is the same.

**Comment by dubeaud** [comment]
You should see more information in the logs now if something is happening.

**Comment by thepaleone** [comment]
Unfortunately there's nothing more in the logs. It's still an IIS 401.1 Unauthorized message that' is being returned..

**Comment by dubeaud** [comment]
Do you have anonymous authentication off and windows enabled in IIS?

**Comment by thepaleone** [comment]
Yes. Only Windows Authentication is enabled, all others are disabled.

**Comment by thepaleone** [comment]
Any other ideas? I've tried changing all kinds of settings (searching for basic troubleshooting of Windows Auth on IIS7) to no avail. About the closest I can come is if I remove the <deny users="?" /> from the web.config, then it actually lets me see the site and displays the propper username in the upper right corner; however, it does not add the user to any of the user tables. If I go to my profile and save some changes (it lets me save items per page) then it adds me to the BugNet_UserProfiles table, but none of the others.

**Comment by dubeaud** [comment]
That is all the module really does is add the user to the users table. I am not sure where the 401 would come from when you have the module enabled as it doesn't do any actual authentication. It does use the username and password in the settings for accessing the ad to get properties. If that failed an exception would be thrown. 

The 401 only happens with module enabled?

**Comment by thepaleone** [comment]
I'm not exactly sure what change I made, but I cannot get the 404 any longer. When I enter the credentials it gives me access to the home page with a list of the projects that should be automatically assigned to new users. It even reads the correct user name in the upper right corner; however, no user has been created. When I use a different account in a different browser my user is not in the list. Looking at the 3 user tables in the database confirms the user has not been created.

I still current have Windows Authentication as the only Authentication method enabled in IIS. I also have the AuthenticationModule loading and I have this for Authorization in my web.config:

<authorization>
    <deny users="?" />
</authorization>

**Comment by dubeaud** [comment]
When you removed the comments from the AD http module in the web.config did you do it in the system.webServer subsection?

If you can confirm that first then the next thing I would try is to change the domain value in the settings to just your root domain, or specify a username and password that has read access to your domain.

**Comment by thepaleone** [comment]
Yes, I actually uncommented both just to be sure (even though I know only 1 makes a difference)

I think it's getting closer. I changed the domain/path setting which was an Active Directory setting before (e.g. CN=Users,DC=<private>,DC=<private>) and I already had user credentials with read access. Changing the domain now causes the user to show up in Admin -> User Accounts and it creates records into several tables including: BugNet_UserProfiles, UsersInRoles, Membership, but now I'm getting an error when I try to add the user to a project. It's very generic - Application Error - /Administration/Users/EditUser/65d2cbdc-ac7a-4bba-8498-a8235ec1b2b4

Unfortunately nothing is logged. I don't know why that is. I have the log4net.config file configured for the ADO appender that it came with, I simply put the same connection string in that the main application is using, but there's nothing being tracked.

**Comment by thepaleone** [comment]
A bit more insight. Apparently the process did not add any records to the aspnet_Users table (or any of those membership tables). That seems to be the cause of the most recent failings. The procedures to add a user to a project include looking up the user in that table.

Is there something else that needs to happen to get the data inserted correctly to those tables?

**Comment by thepaleone** [comment]
OK, I think I may have resolved it. It's been quite a while, but I recall some problems upgrading to v 1.5.258 Apparently only part of the SQL update script ran, there must have been something that caused it to abort.

It was a rather large update so I went through until I found the spot that it must have aborted. I manually ran the scripts after that, then audited the 1.6.x updates to make sure nothing was missed.

After doing all that it now creates the account and everything seems to be working as expected. Thanks for all of your help and suggestions.
